charts/amazon-cloudwatch-observability/templates/linux/fluent-bit-configmap.yaml

```diff
@@ -8,195 +8,14 @@ metadata:
-
-  host-log.conf: |
-    [INPUT]
-        Name                tail
-        Tag                 host.dmesg
-        Path                /var/log/dmesg
-        Key                 message
-        DB                  /var/fluent-bit/state/flb_dmesg.db
-        Mem_Buf_Limit       5MB
-        Skip_Long_Lines     On
-        Refresh_Interval    10
-        Read_from_Head      ${READ_FROM_HEAD}
-
-    [INPUT]
-        Name                tail
-        Tag                 host.messages
-        Path                /var/log/messages
-        Parser              syslog
-        DB                  /var/fluent-bit/state/flb_messages.db
-        Mem_Buf_Limit       5MB
-        Skip_Long_Lines     On
-        Refresh_Interval    10
-        Read_from_Head      ${READ_FROM_HEAD}
-
-    [INPUT]
-        Name                tail
-        Tag                 host.secure
-        Path                /var/log/secure
-        Parser              syslog
-        DB                  /var/fluent-bit/state/flb_secure.db
-        Mem_Buf_Limit       5MB
-        Skip_Long_Lines     On
-        Refresh_Interval    10
-        Read_from_Head      ${READ_FROM_HEAD}
-
-    [FILTER]
-        Name                aws
-        Match               host.*
-        imds_version        v2
-
-    [OUTPUT]
-        Name                cloudwatch_logs
-        Match               host.*
-        region              ${AWS_REGION}
-        log_group_name      /aws/containerinsights/${CLUSTER_NAME}/host
-        log_stream_prefix   ${HOST_NAME}.
-        auto_create_group   true
-        extra_user_agent    container-insights
-
+    {{- .Values.containerLogs.fluentBit.config.service  | nindent 4 }}
+    {{- range $key, $val := .Values.containerLogs.fluentBit.config.extraFiles }}
+    @INCLUDE {{ $key }}
+    {{- end }}
   parsers.conf: |
-    [PARSER]
-        Name                syslog
-        Format              regex
-        Regex               ^(?<time>[^ ]* {1,2}[^ ]* [^ ]*) (?<host>[^ ]*) (?<ident>[a-zA-Z0-9_\/\.\-]*)(?:\[(?<pid>[0-9]+)\])?(?:[^\:]*\:)? *(?<message>.*)$
-        Time_Key            time
-        Time_Format         %b %d %H:%M:%S
-
-    [PARSER]
-        Name                container_firstline
-        Format              regex
-        Regex               (?<log>(?<="log":")\S(?!\.).*?)(?<!\\)".*(?<stream>(?<="stream":").*?)".*(?<time>\d{4}-\d{1,2}-\d{1,2}T\d{2}:\d{2}:\d{2}\.\w*).*(?=})
-        Time_Key            time
-        Time_Format         %Y-%m-%dT%H:%M:%S.%LZ
-
-    [PARSER]
-        Name                cwagent_firstline
-        Format              regex
-        Regex               (?<log>(?<="log":")\d{4}[\/-]\d{1,2}[\/-]\d{1,2}[ T]\d{2}:\d{2}:\d{2}(?!\.).*?)(?<!\\)".*(?<stream>(?<="stream":").*?)".*(?<time>\d{4}-\d{1,2}-\d{1,2}T\d{2}:\d{2}:\d{2}\.\w*).*(?=})
-        Time_Key            time
-        Time_Format         %Y-%m-%dT%H:%M:%S.%LZ
+    {{- .Values.containerLogs.fluentBit.config.customParsers  | nindent 4 }}
```

Comment:
is there a reason we split up the parsers and the fluentbit conf specifically? Seems like there's a lot of in-between configs too such as application-log, host-log, etc.

Reply:
I tried to keep it close to the actual fluent-bit helm chart so that it looks familiar for users already using that.
https://github.com/fluent/helm-charts/blob/main/charts/fluent-bit/values.yaml#L367

Hence Ive exposed the service, customParsers and extraFiles as top level attributes. The application-log, host-log etc are specific to the way we have prescribed these files in our public docs in the past. We typically dont have root level inputs, filters and outputs in the main fluent-bit.conf and always take them in as additional/extra files.